### PR TITLE
Add remove safes dialog

### LINKED ISSUE #692: Port "Remove Safe" dialog
Clarifications:

- You should be able to select multiple safes for deletion.
- This dialog should use the same saga that is being used by the "Adding a Safe" dialog.

Image
Image

### PR DESCRIPTION
**Pull request by alicjakujawa**
## Description

- Port RemoveSafeDialog from colony dap
- Added pink variant for checkbox
- Updated main.css to now show weird borders

Screenshot 2023-07-12 at 13 15 02

Resolves #692

### CODE REVIEW

File: src/components/common/Dialogs/RemoveSafeDialog/RemoveSafeDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import Dialog, { DialogProps, ActionDialogProps } from '~shared/Dialog';
+import { ActionHookForm as Form } from '~shared/Fields';
+
+import { ActionTypes } from '~redux/index';
```
Comment (ArmandoGraterol) [inline src/components/common/Dialogs/RemoveSafeDialog/RemoveSafeDialog.tsx:7]: You can shorten this import by removing the `index`

File: src/components/common/Dialogs/RemoveSafeDialog/SafeListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  isChecked: boolean;
+}
+
+const SafeListItem = ({ safe, isChecked }: Props) => {
+  const { formatMessage } = useIntl();
+  const safeNetwork = SAFE_NETWORKS.find(
+    (network) => network.chainId === Number(safe.chainId),
+  );
+  return (
+    <div className={`${styles.main} ${isChecked && styles.checked}`}>
```
Comment (ArmandoGraterol) [inline src/components/common/Dialogs/RemoveSafeDialog/SafeListItem.tsx:36]: We could use the `classnames` library here and write this in a more readable way:
```
classnames(styles.main, { [styles.checked]: isChecked })
```

File: src/components/common/Dialogs/RemoveSafeDialog/SafeListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      />
+      <Avatar
+        placeholderIcon="circle-close"
+        seed={safe.contractAddress.toLowerCase()}
+        title={safe.safeName || safe.contractAddress}
+        size="xs"
+        className={styles.avatar}
+      />
+
+      <span className={`${isChecked ? styles.selectedLabel : styles.label}`}>
```
Comment (ArmandoGraterol) [inline src/components/common/Dialogs/RemoveSafeDialog/SafeListItem.tsx:52]: Same here

File: src/components/common/Dialogs/RemoveSafeDialog/RemoveSafeDialogForm.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { FormattedMessage, defineMessages } from 'react-intl';
+import { useFormContext } from 'react-hook-form';
+
+import Button from '~shared/Button';
+import DialogSection from '~shared/Dialog/DialogSection';
```
Comment (ArmandoGraterol) [inline src/components/common/Dialogs/RemoveSafeDialog/RemoveSafeDialogForm.tsx:6]: You can also export this section like this `import { DialogSection } from '~shared/Dialog';`, that way we avoid repeating the `DialogSection` at the end of the import route

File: src/components/common/Dialogs/RemoveSafeDialog/RemoveSafeDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  const transform = pipe(
+    mapPayload(({ safeList, annotation: annotationMessage }) => {
+      return {
+        colonyName: colony.name,
+        colonyAddress,
+        safeList,
+        annotationMessage,
+        isRemovingSafes: true,
+      };
+    }),
```
Comment (ArmandoGraterol) [inline src/components/common/Dialogs/RemoveSafeDialog/RemoveSafeDialog.tsx:59]: Normally we extract this function that gets the payload for readability's sake. You can check out the other action dialogs for an example

File: src/components/common/Dialogs/RemoveSafeDialog/RemoveSafeDialogForm.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                  key={`${item.chainId}-${item.contractAddress}`}
+                  safe={item}
+                  isChecked={
+                    !!safes?.find(
+                      (safe) => {
+                        const {contractAddress, chainId} = JSON.parse(safe);
+                        return item.contractAddress === contractAddress &&
+                        item.chainId === chainId
+                      }
+                        ,
```
Comment (ArmandoGraterol) [inline src/components/common/Dialogs/RemoveSafeDialog/RemoveSafeDialogForm.tsx:77]: I'm getting lint errors from this snippet of code

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
Good start! Overall looks good, we just need to address some small stuff and this should be good to go.

- When I get to the dialog, the submit button is already enabled. Also, it stays enabled if I select and then unselect safes. We should check if the form is dirty and has a value before we allow the user to submit it.
FireShot Capture 120 - Colony CDapp - localhost

**Review by ArmandoGraterol — APPROVED** [review]
Looks good to me! Nice job Alicja 💯

**Review by thomassdk — APPROVED** [review]
Works well for me too, nice work Alicija

**Pull request merged**